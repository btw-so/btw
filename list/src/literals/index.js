import { keyMirror } from "@gilbarbara/helpers";

// import { Status } from "types";

export const ActionTypes = keyMirror({
  GITHUB_GET_REPOS_REQUEST: undefined,
  GITHUB_GET_REPOS_SUCCESS: undefined,
  GITHUB_GET_REPOS_FAILURE: undefined,
  HIDE_ALERT: undefined,
  SHOW_ALERT: undefined,
  SET_APP_OPTIONS: undefined,
  USER_LOGIN_REQUEST: undefined,
  USER_LOGIN_SUCCESS: undefined,
  USER_LOGIN_FAILURE: undefined,
  USER_LOGOUT_REQUEST: undefined,
  USER_LOGOUT_SUCCESS: undefined,
  USER_LOGOUT_FAILURE: undefined,
  GET_USER: undefined,
  GET_USER_SUCCESS: undefined,
  GET_USER_FAILURE: undefined,
  GENERATE_OTP: undefined,
  GENERATE_OTP_SUCCESS: undefined,
  GENERATE_OTP_FAILURE: undefined,
  VERIFY_OTP: undefined,
  VERIFY_OTP_SUCCESS: undefined,
  VERIFY_OTP_FAILURE: undefined,
  UPDATE_USER: undefined,
  UPDATE_USER_SUCCESS: undefined,
  UPDATE_USER_FAILURE: undefined,
  RESET_USER_STATE: undefined,
  GET_LIST: undefined,
  GET_LIST_SUCCESS: undefined,
  GET_LIST_FAILURE: undefined,
  NEW_LIST_NODE: undefined,
  EDIT_LIST_NODE_POS: undefined,
  EDIT_LIST_NODE_CONTENT: undefined,
  UPSERT_LIST_NODE: undefined,
  CHANGE_SELECTED_NODE: undefined,
  BATCH_PUSH_NODES: undefined,
  SAVE_NOTE_CONTENT: undefined,
  GET_NOTE: undefined,
  GET_NOTE_SUCCESS: undefined,
  GET_NOTE_FAILURE: undefined,
  ADD_FILE: undefined,
  ADD_FILE_SUCCESS: undefined,
  ADD_FILE_FAILURE: undefined,
  GET_FILE: undefined,
  GET_FILE_SUCCESS: undefined,
  GET_FILE_FAILURE: undefined,
  GET_PINNED_NODES: undefined,
  GET_PINNED_NODES_SUCCESS: undefined,
  GET_PINNED_NODES_FAILURE: undefined,
  GET_PUBLIC_NOTE: undefined,
  GET_PUBLIC_NOTE_SUCCESS: undefined,
  GET_PUBLIC_NOTE_FAILURE: undefined,
  SEARCH_NODES: undefined,
  SEARCH_NODES_SUCCESS: undefined,
  SEARCH_NODES_FAILURE: undefined,
});

export const STATUS = {
  IDLE: "idle",
  RUNNING: "running",
  READY: "ready",
  SUCCESS: "success",
  ERROR: "error",
  PINNED: "pinned",
};
