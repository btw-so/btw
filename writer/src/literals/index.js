import { keyMirror } from "@gilbarbara/helpers";

// import { Status } from "types";

export const ActionTypes = keyMirror({
  GITHUB_GET_REPOS_REQUEST: undefined,
  GITHUB_GET_REPOS_SUCCESS: undefined,
  GITHUB_GET_REPOS_FAILURE: undefined,
  HIDE_ALERT: undefined,
  SHOW_ALERT: undefined,
  SET_APP_OPTIONS: undefined,
  USER_LOGIN_REQUEST: undefined,
  USER_LOGIN_SUCCESS: undefined,
  USER_LOGIN_FAILURE: undefined,
  USER_LOGOUT_REQUEST: undefined,
  USER_LOGOUT_SUCCESS: undefined,
  USER_LOGOUT_FAILURE: undefined,
  GET_USER: undefined,
  GET_USER_SUCCESS: undefined,
  GET_USER_FAILURE: undefined,
  GENERATE_OTP: undefined,
  GENERATE_OTP_SUCCESS: undefined,
  GENERATE_OTP_FAILURE: undefined,
  VERIFY_OTP: undefined,
  VERIFY_OTP_SUCCESS: undefined,
  VERIFY_OTP_FAILURE: undefined,
  UPDATE_USER: undefined,
  UPDATE_USER_SUCCESS: undefined,
  UPDATE_USER_FAILURE: undefined,
  GET_NOTES: undefined,
  GET_NOTES_SUCCESS: undefined,
  GET_NOTES_FAILURE: undefined,
  UPSERT_NOTE: undefined,
  UPSERT_NOTE_SUCCESS: undefined,
  UPSERT_NOTE_FAILURE: undefined,
  CREATE_NEW_NOTE: undefined,
  SELECT_NOTE: undefined,
  SAVE_NOTE_CONTENT: undefined,
  IMPORT_NOTES: undefined,
  ADD_CUSTOM_DOMAIN: undefined,
  ADD_CUSTOM_DOMAIN_SUCCESS: undefined,
  ADD_CUSTOM_DOMAIN_FAILURE: undefined,
  PUBLISH_NOTE: undefined,
  PUBLISH_NOTE_SUCCESS: undefined,
  PUBLISH_NOTE_FAILURE: undefined,
  ARCHIVE_NOTE: undefined,
  ARCHIVE_NOTE_SUCCESS: undefined,
  ARCHIVE_NOTE_FAILURE: undefined,
  DELETE_NOTE: undefined,
  DELETE_NOTE_SUCCESS: undefined,
  DELETE_NOTE_FAILURE: undefined,
  GET_NOTE: undefined,
  GET_NOTE_SUCCESS: undefined,
  GET_NOTE_FAILURE: undefined,
  RESET_USER_STATE: undefined,
  SET_NOTE_SLUG: undefined,
  SET_NOTE_SLUG_SUCCESS: undefined,
  SET_NOTE_SLUG_FAILURE: undefined,
  SEARCH_NOTES: undefined,
  SEARCH_NOTES_SUCCESS: undefined,
  SEARCH_NOTES_FAILURE: undefined,
});

export const STATUS = {
  IDLE: "idle",
  RUNNING: "running",
  READY: "ready",
  SUCCESS: "success",
  ERROR: "error",
};
