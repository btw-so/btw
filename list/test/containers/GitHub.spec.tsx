import React from 'react';

import { ActionTypes } from 'literals';

import GitHub from 'containers/GitHub';

import githubRepos from 'test/__fixtures__/github-repos.json';
import { fireEvent, render, screen, waitFor } from 'test-utils';

const mockDispatch = jest.fn();

describe('GitHub', () => {
  afterEach(() => {
    mockDispatch.mockClear();
    fetchMock.resetMocks();
  });

  it('should render a loader and dispatch the action', () => {
    const { container } = render(<GitHub />, { mockDispatch });

    expect(container).toMatchSnapshot();
    expect(mockDispatch).toHaveBeenCalledWith({
      type: ActionTypes.GITHUB_GET_REPOS_REQUEST,
      payload: 'react',
    });
  });

  it('should render the repos', async () => {
    fetchMock.mockResponse(JSON.stringify({ items: githubRepos.items.slice(0, 2) }));

    render(<GitHub />);
    const repos = await screen.findByTestId('GitHubGrid');

    expect(repos.outerHTML).toMatchSnapshot();
  });

  it("should render a message if there's no data", async () => {
    fetchMock.mockResponse(JSON.stringify({ items: [] }));

    render(<GitHub />);

    const repos = await screen.findByText('Nothing found');

    expect(repos).toMatchSnapshot();
  });

  it('should dispatch an action when click selector button', async () => {
    fetchMock.mockResponse(JSON.stringify({ items: [] }));
    render(<GitHub />, { mockDispatch });

    fireEvent.click(screen.getByRole('button', { name: 'Redux' }));

    expect(mockDispatch).toHaveBeenCalledWith({
      type: 'GITHUB_GET_REPOS_REQUEST',
      payload: 'redux',
    });
  });

  it('should dispatch an alert with errors', async () => {
    fetchMock.mockReject(new Error('Nothing found'));

    render(<GitHub />, { mockDispatch });

    await waitFor(() => {
      expect(mockDispatch).toHaveBeenCalledWith({
        type: 'SHOW_ALERT',
        payload: {
          id: '8cdee72f-28d4-4441-91f0-c61f6e3d9684',
          icon: 'dot-circle-o',
          message: 'Nothing found',
          position: 'bottom-right',
          variant: 'danger',
          timeout: 0,
        },
      });
    });
  });
});
