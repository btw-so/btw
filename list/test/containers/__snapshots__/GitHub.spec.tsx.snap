// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitHub should render a loader and dispatch the action 1`] = `
.c3 {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #fff;
  border: 1px solid #4e4141;
  color: #4e4141;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0px;
  box-shadow: none;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 16px;
  line-height: 1;
  padding: 8px 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: auto;
}

.c3:disabled {
  opacity: 0.7;
  pointer-events: none;
}

.c3:focus {
  outline-color: #4e4141;
}

.c2 {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #4e4141;
  border: 1px solid #4e4141;
  color: #fff;
  -webkit-animation: bhbJZt 0.5s linear infinite;
  animation: bhbJZt 0.5s linear infinite;
  background-image: repeating-linear-gradient( 45deg, rgba(255,255,255,0.4), rgba(255,255,255,0.4) 20px, transparent 20px, transparent 40px );
  background-size: 56px 56px;
  pointer-events: none;
  -webkit-transition: none !important;
  transition: none !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0px;
  box-shadow: none;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 16px;
  line-height: 1;
  padding: 8px 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: auto;
}

.c2:disabled {
  opacity: 0.7;
  pointer-events: none;
}

.c2:focus {
  outline-color: #4e4141;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 > button + button {
  margin-left: -1px;
}

.c1 > button:first-of-type {
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
}

.c1 > button:last-of-type {
  border-bottom-left-radius: 0;
  border-top-left-radius: 0;
}

.c1 > button:not(:first-of-type):not(:last-of-type) {
  border-radius: 0;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 32px;
  margin: 2rem auto;
  position: relative;
  width: 32px;
}

.c4 > div {
  -webkit-animation: hqrBsq 1.15s infinite cubic-bezier(0.2,0.6,0.36,1);
  animation: hqrBsq 1.15s infinite cubic-bezier(0.2,0.6,0.36,1);
  border: 0 solid #00b4d5;
  border-radius: 50%;
  box-sizing: border-box;
  height: 0;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 0;
}

<div>
  <div
    data-testid="GitHubWrapper"
  >
    <div
      class="c0"
    >
      <div
        aria-label="GitHub Selector"
        class="c1"
        data-testid="GitHubSelector"
        role="group"
      >
        <button
          class="c2"
          data-value="react"
          type="button"
        >
          React
        </button>
        <button
          class="c3"
          data-value="redux"
          type="button"
        >
          Redux
        </button>
      </div>
    </div>
    <div
      class="c4"
      color="#00b4d5"
      data-testid="Loader"
      size="32"
      type="grow"
    >
      <div />
    </div>
  </div>
</div>
`;

exports[`GitHub should render a message if there's no data 1`] = `
<h3>
  Nothing found
</h3>
`;

exports[`GitHub should render the repos 1`] = `
<div data-testid="GitHubGrid"
     data-value="react"
     width="[object Object]"
     class="sc-bjUoiL fMZXtX"
>
  <a href="https://github.com/freeCodeCamp/freeCodeCamp"
     target="_blank"
     class="sc-cxabCf gJbJSN Item-sc-jn8jzz-0 eHJsUR"
  >
    <img alt="freeCodeCamp"
         src="https://avatars0.githubusercontent.com/u/9892522?v=4"
         class="sc-hHLeRK bOPpni"
    >
    <div class="ItemHeader-sc-jn8jzz-1 ikJHPS">
      <h5 class="sc-dIouRR kcAabS">
        freeCodeCamp
      </h5>
      <small>
        freeCodeCamp
      </small>
    </div>
    <p class="sc-gicCDI RpExH">
      freeCodeCamp.org's open source codebase and curriculum. Learn to code at home.
    </p>
  </a>
  <a href="https://github.com/facebook/react"
     target="_blank"
     class="sc-cxabCf gJbJSN Item-sc-jn8jzz-0 eHJsUR"
  >
    <img alt="facebook"
         src="https://avatars3.githubusercontent.com/u/69631?v=4"
         class="sc-hHLeRK bOPpni"
    >
    <div class="ItemHeader-sc-jn8jzz-1 ikJHPS">
      <h5 class="sc-dIouRR kcAabS">
        react
      </h5>
      <small>
        facebook
      </small>
    </div>
    <p class="sc-gicCDI RpExH">
      A declarative, efficient, and flexible JavaScript library for building user interfaces.
    </p>
  </a>
</div>
`;
